docs/db_dev_process/new_fuzzer.rst

```diff
@@ -0,0 +1,164 @@
+2)) =\> 8 bitstreams (specimens) to solve all of them.
+
+Calculating the base address
+++++++++++++++++++++++++++++
+
+#.  Find a tilegrid fuzzer to copy, e.g. "dsp"
+
+#.  Enter your copied directory
+
+#.  Edit `top.py`
+
+    a.  Refer to the `Xilinx 7 Series Library guide <https://www.xilinx.com/support/documentation/sw_manuals/xilinx2012_2/ug953-vivado-7series-libraries.pdf>`_ and/or Vivado layout to understand the primitive you need to instantiate
+
+    b.  Find a single bit parameter that can be easily toggled, such as a clock inverter or a bulk configuration bit
+
+    c.  Find the correct site type in gen_sites()
+
+    d.  Instantiate the correct verilog library macro in top
+
+    e.  LOC it, if necessary. It's necessary to LOC it if there is more than one
+
+#.  Run make, and look at Vivado's output. Especially if you took shortcuts instantiating your macro (ex: not connecting critical ports) you may need to add DRC waivers to generate.tcl
+
+#.  Inspect the ``build/segbits_tilegrid.tdb`` to observe bit addresses, for example ``DSP_L_X22Y0.DWORD:0.DFRAME:1b 0040171B_000_01``
+
+    #.  The ``DFRAME`` etc entries are deltas to convert this feature offset to the base address for the tile
+
+    #.  We will fix them in the subsequent step
+
+#.  Correct Makefile's ``GENERATE_ARGS`` to make it the section base address instead of a specific bit in that memory region
+
+    #.  Align address to 0x80: 0x0040171B => --dframe 1B to yield a base address of 0x00401700
+
+    #.  Correct word offset. This is harder since it requires some knowledge of how and where the IP block memory is as a whole
+
+        i.  If there is only one tile of this type in the DSP column:
+            start by assuming it occupies the entire address range.
+            In this step add a delta to make the word offset 0 (--dword 0) and later indicate that it occupies 101 words (all of them)
+
+        ii. If there are multiple: compare the delta between adjacent tiles to get the pitch.
+            This should give an upper bound on the address size.
+            Make a guess with that in mind and you may have to correct it later when you have better information.
+
+    #.  Align bits to 0: 1 => --dbit 1
+
+#.  Run ``make clean && make``
+
+#.  Verify ``build/segbits_tilegrid.tdb`` now looks resolved
+
+    #.  Ex: ``DSP_L_X22Y0.DWORD:0.DFRAME:1b 0040171B_000_01``
+
+    #.  In this case there were several DSP48 sites per DSP column
+
+#.  Find the number of frames for your tile
+
+    #.  Run ``$XRAY_BLOCKWIDTH build/specimen_001/design.bit``
+
+    #.  Find the base address you used above i.e. we used ``0x00401700``, so use ``0x00401700: 0x1B`` (0x1C => 28)
+
+    #.  This information is in the part YAML file, but is not as easy to read
+
+#. Return to the main tilegrid directory
+
+#. Edit ``tilegrid/add_tdb.py`` dsd a
+
+   #.  Find ``tdb\_fns`` and add an entry for your tile type e.g. ``(dsp/build/segbits_tilegrid.tdb", 28, 10)``
```

Comment:
Don't need to escape underscores that are fixed width?

Reply:
Fixed